src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts

```diff
@@ -510,22 +511,29 @@ export function* updateRepeatingGroupEditIndexSaga({
   payload: { group, index, validate },
 }: PayloadAction<IUpdateRepeatingGroupsEditIndex>): SagaIterator {
   try {
-    if (validate) {
-      const state: IRuntimeState = yield select();
+    const state: IRuntimeState = yield select();
+    const rowIndex = state.formLayout.uiConfig.repeatingGroups?.[group].editIndex;
+
+    if (validate && typeof rowIndex === 'number' && rowIndex > -1) {
       const validations: IValidations = state.formValidations.validations;
       const currentView = state.formLayout.uiConfig.currentView;
-      const rowIndex = state.formLayout.uiConfig.repeatingGroups?.[group].editIndex;
 
       const frontendValidations: IValidations = validateGroup(
         group,
         state,
         validate === Triggers.ValidateRow ? rowIndex : undefined,
       );
 
+      // Get group's rowIndices to send to server for validations
+      const nodes = nodesInLayouts(state.formLayout.layouts, currentView, state.formLayout.uiConfig.repeatingGroups);
+      const groupNode = nodes.findById(group);
+      const rowIndices = groupNode?.rowIndices();
+      rowIndices?.push(rowIndex);
```

Comment:
The `group` variables already contains the index (embedded inside the string). You can extract it by using the `splitDashedKey` function without constructing the entire layout.

Reply:
Ait nice, did not realize this function existed. It does have the potential for introducing weird bugs though since the `componentId` can be anything at all, but I suppose that case is rare enough. [This issue](https://github.com/Altinn/app-frontend-react/issues/345) keeps coming to mind when working with validations 😉

Reply:
Yep - but it's been that way for a long time. We should absolutely get away from it long-term, but short-term I think we should consider running a [sanity check](https://github.com/Altinn/app-frontend-react/issues/648) and warn if component IDs can be mis-interpreted as containing row indexes. (See the task starting with _As long as app-frontend-react puts the repeating group indexes into the component ID_ in the linked issue)